examples/lighting-app/efr32/args.gni

```diff
@@ -13,6 +13,20 @@
 # limitations under the License.
 
 import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
 import("${chip_root}/examples/platform/efr32/args.gni")
 
 efr32_sdk_target = get_label_info(":sdk", "label_no_toolchain")
+
+declare_args() {
+  chip_build_pw_rpc_lib = false
+  dir_pw_log_chip = "$chip_root/src/lib/support/pw_log_chip"
+}
+
+if (chip_build_pw_rpc_lib) {
```

Comment:
This won't work, it will always be false because args.gni is evaluated before the args are decided (it sets the defaults, cannot use args before they are set). I think you'll need to have multiple files

args.gni
args_pw_rpc.gni

and import one or the other into the command line.

Reply:
So when is the value of arguments from the command line actually applied to the declare_args() ?

Is there not a way to set build configuration base on argument variable?

Reply:
It's applied to declare_args() in all normal build files right when declare_args() is evaluated, the problem is that args.gni is _not_ a normal build file. It's a special build file contains defaults for arguments.

The default arguments is a single scope and cannot have conditions embedded in it. So we have to do something else, like have multiple args.gni and configure variants like this

`gn gen --args='import("//args/with_pw_rpc.gni")'`

```
// with_pw_rpc.gni
pw_log_BACKEND = "$dir_pw_log_basic"
pw_assert_BACKEND = "$dir_pw_assert_log"
```

Reply:
I did the modification and added a with_pw_rpc.gni to the three EFR32 example app. Those can be imported in args when we will want to build one of these examples with pw_rpc. 

pw_rpc for lock and light examples is still a WiP. I need to bridge standard logs to the pw_logs. This is my next step once the pigweed-app example is merged.